tfhe/src/integer/gpu/server_key/radix/scalar_div_mod.rs

```diff
@@ -0,0 +1,394 @@
+                self.unchecked_scalar_right_shift_assign_async(&mut quotient, 1, streams);
+            }
+            self.add_assign(&mut quotient, &t1, streams);
+            assert!(chosen_multiplier.shift_post > 0);
+
+            unsafe {
+                self.unchecked_scalar_right_shift_assign_async(
+                    &mut quotient,
+                    chosen_multiplier.shift_post as u64 - 1,
+                    streams,
+                );
+            }
+            streams.synchronize();
+            quotient
+        } else {
+            let shifted_n = self.unchecked_scalar_right_shift(numerator, shift_pre, streams);
+            let mut quotient =
+                self.scalar_mul_high(&shifted_n, chosen_multiplier.multiplier, streams);
+            unsafe {
+                self.unchecked_scalar_right_shift_assign_async(
+                    &mut quotient,
+                    chosen_multiplier.shift_post as u64,
+                    streams,
+                );
+            }
+            streams.synchronize();
+            quotient
+        }
+    }
+
+    /// Computes homomorphically a division between a ciphertext and a scalar.
+    ///
+    /// This function computes the operation without checking if it exceeds the capacity of the
+    /// ciphertext.
+    ///
+    /// The result is returned as a new ciphertext.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::core_crypto::gpu::CudaStreams;
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::integer::gpu::ciphertext::CudaUnsignedRadixCiphertext;
+    /// use tfhe::integer::gpu::gen_keys_radix_gpu;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// let gpu_index = 0;
+    /// let mut streams = CudaStreams::new_single_gpu(gpu_index);
+    ///
+    /// // We have 4 * 2 = 8 bits of message
+    /// let size = 4;
+    /// let (cks, sks) = gen_keys_radix_gpu(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size, &mut streams);
+    ///
+    /// let msg = 30;
+    /// let scalar = 3;
+    ///
+    /// let ct = cks.encrypt(msg);
+    /// let mut d_ct = CudaUnsignedRadixCiphertext::from_radix_ciphertext(&ct, &mut streams);
+    ///
+    /// // Compute homomorphically a scalar multiplication:
+    /// let d_ct_res = sks.scalar_div(&d_ct, scalar, &mut streams);
+    /// let ct_res = d_ct_res.to_radix_ciphertext(&mut streams);
+    ///
+    /// let clear: u64 = cks.decrypt(&ct_res);
+    /// assert_eq!(msg / scalar, clear);
+    /// ```
+    pub fn scalar_div<Scalar>(
+        &self,
+        numerator: &CudaUnsignedRadixCiphertext,
+        divisor: Scalar,
+        streams: &CudaStreams,
+    ) -> CudaUnsignedRadixCiphertext
+    where
+        Scalar: Reciprocable,
+    {
+        let mut result = unsafe { numerator.duplicate_async(streams) };
+        if !numerator.block_carries_are_empty() {
+            unsafe {
+                self.full_propagate_assign_async(&mut result, streams);
+            }
+        };
```

Comment:
I think it should be

``` rust
let mut tmp_numerator;
let numerator = if numerator.block_carries_are_empty() {
  numerator
} else {
    unsafe {
        tmp_numerator = numerator.duplicate_async(streams);
       self.full_propagate_assign_async(&mut tmp_numerator, streams);
    }
  &tmo_numerator
}
```

Its closer to what we do on CPU and allows to only copy the input in its carries are not clean